distribution/integration/tests-integration/src/test/java/org/wso2/carbon/integration/tests/carbontools/CarbonServerBasicOperationTestCase.java

```diff
@@ -0,0 +1,219 @@
+import org.wso2.carbon.automation.engine.context.ContextXpathConstants;
+import org.wso2.carbon.integration.common.utils.CarbonCommandToolsUtil;
+import org.wso2.carbon.integration.common.utils.CarbonIntegrationBaseTest;
+import org.wso2.carbon.utils.ServerConstants;
+import sun.management.VMManagement;
+
+import java.io.File;
+import java.lang.management.ManagementFactory;
+import java.lang.management.RuntimeMXBean;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.HashMap;
+
+import static org.testng.Assert.assertTrue;
+
+/**
+ * This class has the test methods for start, stop, restart, dump and run build.xml test cases
+ */
+public class CarbonServerBasicOperationTestCase extends CarbonIntegrationBaseTest {
+
+    private static final Log log = LogFactory.getLog(CarbonServerBasicOperationTestCase.class);
+    private HashMap<String, String> serverPropertyMap = new HashMap<String, String>();
+    private String carbonHome;
+    private AutomationContext context;
+    private final int portOffset = 1;
+    private String processId;
+    Process processStop;
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+        serverPropertyMap.put("-DportOffset", Integer.toString(portOffset));
+        context = new AutomationContext("CARBON", "carbon002",
+                                        ContextXpathConstants.SUPER_TENANT,
+                                        ContextXpathConstants.ADMIN);
+        carbonHome = CarbonCommandToolsUtil.getCarbonHome(context);
+    }
+
+    @Test(groups = {"wso2.as"}, description = "Server start test")
+    public void testStartCommand() throws Exception {
+        String[] cmdArrayToStart;
+        Process process;
+
+        if (CarbonCommandToolsUtil.isCurrentOSWindows()) {
+            throw new SkipException("Feature --start is not available for windows");
+        } else {
+            cmdArrayToStart = new String[]
+                    {"sh", "wso2server.sh", "--start", "-DportOffset=" + portOffset};
+            process = CarbonCommandToolsUtil.runScript(carbonHome + "/bin", cmdArrayToStart);
+        }
+        boolean startupStatus = CarbonCommandToolsUtil.isServerStartedUp(context, portOffset);
+        // Waiting until start the server
+
+        Field field = process.getClass().getDeclaredField("pid");
+        field.setAccessible(true);
+        processId = field.get(process).toString();
+        log.info("process id for carbon server with offset 1 : " + processId);
+
+        assertTrue(startupStatus, "Unsuccessful login");
+    }
+
+    @Test(groups = {"wso2.as"}, description = "Server start test", dependsOnMethods = {"testStartCommand"})
+    public void testDumpCommandOnLinux() throws Exception {
+        String[] cmdArray;
+        Process processDump = null;
+        try {
+            if (CarbonCommandToolsUtil.isCurrentOSWindows()) {
+                throw new SkipException("Feature --start is not available for windows");
```

Comment:
get utill to return current operating system and make change according to returning value
CarbonCommandToolsUtil.getCurrentOperatingSystem()

if(CarbonCommandToolsUtil.getCurrentOperatingSystem().equals(OperatingSystem.WINDOWS))